### DISCUSSION TITLE
Run parameters are not getting picked up in plugin tag.

### DISCUSSION THREAD

**Discussion by blevinson** [artilleryio/artillery#1687] [Artillery]
I'm trying to dynamically change the test_id so it reflects well in the graph. The tag in the test changes but on in the plugin. How can I get the plugin to pull in the "token" variable.

Test:
config:
  target: "https://site.com"
  plugins:
    publish-metrics:
      - type: prometheus
        pushgateway: "http://[IP]:9091"
        prefix: 'artillery_k8s'
        tags:
          - "test_id:test-{{ token }}"  <-- this won't change
          - "type:loadtest"
...

before:
  flow:
    - log: "Running: test-{{ token }}". <-- this will

Command:
TEST_ID=$(echo $RANDOM | md5sum | head -c 5; echo;)
npx artillery run scenario.yaml --variables "{\"token\":\"$TEST_ID\"}"

Please assist as soon as possible

**Comment by hassy** [comment]
hi @blevinson 👋 

Variables set with the `--variables` flag are only available inside scenario definitions.

Environment variables may be used anywhere in the script and can be accessed through `$processEnvironment`, e.g. the `TEST_ID` environment variable is under  `$processEnvironment.TEST_ID`.

**Comment by blevinson** [comment]
Understood, thank you!

**Comment by estufamagica** [comment]
Hey, I got an issue that is pretty similar. 

I got my testId for influx set to a environment variable to change its value dynamically, the thing is I haven't been able to change it in execution time. 

  plugins:
    publish-metrics:
      - type: influxdb-statsd # https://artillery.io/docs/guides/plugins/plugin-publish-metrics.html#InfluxDB-Telegraf
        tags: { testId: "{{ $env.SERVICE }}", reporterType: influxdb-statsd}
        event:
          priority: low
scenarios:
  # We define one scenario:
  - name: "DSLogsRequests"
    flow:
      #- log: "Service: {{ service }} - JSON: {{ json }}"
      - post:
          url: "/data_service"
          beforeRequest: setEnvironmentVariable
          body: "{{ json }}"
          headers:
            content-type: application/json

The {{ json }} is a parameter from a csv where I got the json bodies for the request. The thing is I am testing a single endpoint but it has multiple services I have to test. That's why i need the service field, for grouping results in grafana. 

setEnvironmentVariable is a js function that changes the value of the environment variable. However the value of testId dosen't seem to change at all. 

{{ service }} ->the csv of the json body also has a field for the service name. 

Is there a way to change dynamically the value in execution time?
